btcutil/psbt/partial_input.go

```diff
@@ -328,6 +483,85 @@ func (pi *PInput) serialize(w io.Writer) error {
 				return err
 			}
 		}
+
+		if pi.TaprootKeySpendSig != nil {
+			err := serializeKVPairWithType(
+				w, uint8(TaprootKeySpendSignatureType), nil,
+				pi.TaprootKeySpendSig,
+			)
+			if err != nil {
+				return err
+			}
+		}
+
+		sort.Sort(TaprootScriptSpendSigSorter(pi.TaprootScriptSpendSig))
```

Comment:
Where does the BIP actually mandate that these contents be sorted? Or is this just something we're adding to make testing easier since our serialization would be deterministic-ish?